core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -302,6 +302,53 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def setBillingProjectSpendConfiguration(projectName: RawlsBillingProjectName, datasetName: String): Future[Int] = {
+
+    validateBigQueryDatasetName(datasetName)
+
+    requireProjectAction(projectName, SamBillingProjectActions.alterSpendReportConfiguration) {
+      val bqService = bqServiceFactory.getServiceFromCredentialPath(pathToCredentialJson, GoogleProject(projectName.value))
+
+      for {
+        //Get the dataset to validate that it exists and that we have permission to see it
+        _ <- bqService.use(_.getDataset(datasetName)).unsafeToFuture().map {
+          case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The dataset $datasetName could not be found."))
+          case dataset => dataset
+        }
+
+        //The billing account ID in the DB can't be trusted, so we'll get it directly from the source of truth
+        billingAccountIdOpt <- gcsDAO.getProjectBillingAccount(projectName, userInfo)
+
+        //Get the parent billing account for the projects billing account
+        rootBillingAccountId <- billingAccountIdOpt match {
+                                    case Some(billingAccountId) => gcsDAO.getRootBillingAccount(billingAccountId, userInfo).map(_.getOrElse(billingAccountId))
+                                    case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The billing project ${projectName.value} is not linked to an active billing account."))
+                                  }
+
+        //Get the table and validate that it exists and that we have permission to see it
+        tableName = s"gcp_billing_export_v1_${rootBillingAccountId.stripPrefix("billingAccounts/")}"
```

Comment:
Apologies for not having read all the Google docs on this yet, but is this ALWAYS the case?  That even if I configure the BigQuery data export for my subaccount, the spend data will end up in a table named after the Root Billing Account?